<?xml version="1.0" encoding="utf-8"?>
<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="color_format" value="yuv422p" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <!-- <param name="camera_info_url" type="string" value="$(find yolov5_ros)/config/ost.yaml"/>  -->
  </node>

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>  <!--/camera/color/image_raw--> <!--/usb_cam/image_raw-->
    <param name="autosize" value="true" />
  </node>

  <param name="yolov5_path"       value="$(find yolov5_ros)/yolov5"/>
  <param name="use_cpu"           value="false" />
    <node pkg="yolov5_ros" type="yolo_main.py" name="yolov5_ros" output="screen" >
    <param name="weight_path"       value="$(find yolov5_ros)/weights/best3_200.pt"/>
    <param name="image_topic"       value="/usb_cam/image_raw" />
    <param name="pub_topic"         value="BoundingBoxes" />
    <param name="camera_frame"       value="camera_color_frame"/>
    <param name="conf"              value="0.65" />   <!--0.15-->
    <param name="iou"               value="0.7" />    <!--0.6-->
  </node>
  
  <node pkg="yolov5_dection" type="dection_basket" name="dection_basket" output="screen"/>
  <!-- <node pkg="yolov5_drive" type="descion" name="descion" output="screen"/> -->
 </launch>